src/Microsoft.AspNetCore.Razor.Evolution/RazorSyntaxTree.cs

```diff
@@ -32,11 +32,10 @@ public static RazorSyntaxTree Parse(RazorSourceDocument source)
             }
 
             var parser = new RazorParser();
+            var sourceContent = new char[source.Length];
+            source.CopyTo(0, sourceContent, 0, source.Length);
 
-            using (var reader = source.CreateReader())
-            {
-                return parser.Parse(reader);
-            }
+            return parser.Parse(sourceContent);
```

Comment:
It would be even better to see the tokenizer use `SourceDocument` directly. May not be feasible in one move